crates/crypto/src/hash/mod.rs

```diff
@@ -39,12 +41,32 @@ impl FromStr for HashAlgorithm {
     }
 }
 
-pub trait SupportedDigest: Digest + private::Sealed {
+static EMPTY_HASH_CACHE: Lazy<HashMap<u8, Hash<Sha256>>> = Lazy::new(|| {
+    let mut m: HashMap<u8, Hash<Sha256>> = HashMap::new();
```

Comment:
There's no need for a hashmap since this will always be built up sequentially. Use a `Vec` instead and simply build it up iteratively with roughly `hashes.push(branch_hash(last, last))`.